### PR TITLE
Add more symbols to the import hinter

### PR DESCRIPTION
**Pull request by wilzbach**
A small start with a few of the more popular symbols from Phobos.

(from https://forum.dlang.org/post/q8eulj$2hin$1@digitalmars.com)

### CODE REVIEW

**Review by jacob-carlborg — APPROVED** [review]

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/imphint.d [hunk 1]
```diff
@@ -34,15 +34,49 @@ shared static this()
...
+        "popFront": "std.range",
         "printf": "core.stdc.stdio",
         "realloc": "core.stdc.stdlib",
+        "replace": "std.array",
+        "rmdir": "std.file",
         "sin": "std.math",
+        "sort": "std.algorithm",
+        "split": "std.array",
         "sqrt": "std.math",
+        "startsWith": "std.algorithm",
```
Comment (Geod24) [inline src/dmd/imphint.d]: `std.algorithm.searching`?

**Comment by Geod24** [comment]
Actually shouldn't those `std.algorithm` be more specific ?

**Comment by wilzbach** [comment]
I actually did think about this for a bit (same for the std.range imports), but it boils down to: you don't save any import time as the entire package gets parsed by DMD anyhow (as it and so many other packages import it). Two years ago or so, there was a move to make lots of imports of std.algorithm more specialized with little to no impact. On the other hand it increases the overhead to the user who has to remember more packages. As this is more targeted at newcomers who might have forgotten the import, I think we should provide them with the easiest solution to remember.

**Comment by Geod24** [comment]
@wilzbach : It has little to do with import time in my case, but with finer-grained dependencies.

**Comment by wilzbach** [comment]
What's the benefit for the user?

FWIW with the rise of `import std`, I think the trend goes the other way.

Anyhow, another reason is less typing which is why I presume selective imports aren't suggested by imphint either.

**Comment by Geod24** [comment]
> FWIW with the rise of import std, I think the trend goes the other way.

Then just get rid of `imphint` altogether.

I see it as a difference in approach: yours seems more geared towards scripts, mine towards larger programs. In your framework it makes sense, but on the other hand I'm a bit wary of having the compiler recommend importing `std.algorithm` as my understanding was that's something we were not recommending in general.

**Pull request merged**